Extended Kotlin DSL

gmariotti opened the pull request:
The purpose of this pull request is to improve the Kotlin experience when using the consumer DSL. Differently from the previous `LambdaDslJsonArray.newObject`, the new methods avoid the repetition of `it` inside the lambda by using, as a parameter, a lambda with receiver.

Questions:
- What is the best way to test these extensions?
- What kind of documentation is it needed (JavaDoc, README examples, ...)

Thank you in advance for investing your time in looking into it.

uglyog commented:
Thanks, this looks useful. To test it further than what you did is to write a consumer test, maybe in pact-jvm-consumer-junit or pact-jvm-consumer-junit5. But that's not too important, your test shows how it works.

For documentation, just add a section to the README with an example of how to use it with Kotlin.

gmariotti commented:
Hi @uglyog, I've updated the pull request with some documentation and an example in the DSL README. I've also added a couple of extension methods for `LambdaDslJsonArray`, however, I had to change the signature of `LambdaDslJsonArray.newObject` so that `this` can be used instead of `it`. Is this a problem?

uglyog commented:
All good. Thanks!

gmariotti commented:
Thank you for accepting it!!
